bookkeeper-server/src/test/java/org/apache/bookkeeper/bookie/BookieJournalTest.java

```diff
@@ -794,4 +801,53 @@ private void testPartialFileInfoPostV3Journal(boolean truncateMasterKey)
             // correct behaviour
         }
     }
+
+    @Test(expected = IOException.class)
+    public void testJournalScanIOException() throws Exception {
+        File journalDir = createTempDir("bookie", "journal");
+        Bookie.checkDirectoryStructure(Bookie.getCurrentDirectory(journalDir));
+
+        File ledgerDir = createTempDir("bookie", "ledger");
+        Bookie.checkDirectoryStructure(Bookie.getCurrentDirectory(ledgerDir));
+
+        writeV4Journal(Bookie.getCurrentDirectory(journalDir), 100, "testPasswd".getBytes());
+
+        ServerConfiguration conf = TestBKConfiguration.newServerConfiguration();
+        conf.setJournalDirName(journalDir.getPath())
+                .setLedgerDirNames(new String[] { ledgerDir.getPath() })
+                .setMetadataServiceUri(null);
+
+        Journal.JournalScanner journalScanner = new DummyJournalScan();
+        FileChannel fileChannel = PowerMockito.mock(FileChannel.class);
+
+        PowerMockito.when(fileChannel.position(Mockito.anyLong()))
+                .thenThrow(new IOException());
+
+        PowerMockito.mockStatic(JournalChannel.class);
+        PowerMockito.when(JournalChannel.openFileChannel(Mockito.any(RandomAccessFile.class))).thenReturn(fileChannel);
+
+        Bookie b = null;
+
+        try {
+            b = new Bookie(conf);
+        } finally {
+            b.shutdown();
+        }
+
+        for (Journal journal : b.journals) {
+            List<Long> journalIds = journal.listJournalIds(journal.getJournalDirectory(), null);
+
+            assertEquals(journalIds.size(), 1);
+
+            journal.scanJournal(journalIds.get(0), Long.MAX_VALUE, journalScanner);
```

Comment:
try {
    journal.scanJournal(journalIds.get(0), Long.MAX_VALUE, journalScanner) ;
   fail("should not be able to scan the journal")
}  catch (IOException err) {
   // expected
   // maybe you can perform some assertion about the err.getMessage() result, but maybe it depends on java version or locale
}

Reply:
I tried that but it seems too brittle a check for depending on the error message.